src/librbd/api/Trash.cc

```diff
@@ -163,7 +163,7 @@ int Trash<I>::move(librados::IoCtx &io_ctx, rbd_trash_image_source_t source,
                  << dendl;
 
   auto ictx = new I("", image_id, nullptr, io_ctx, false);
-  int r = ictx->state->open(OPEN_FLAG_SKIP_OPEN_PARENT);
+  int r = ictx->state->open(OPEN_FLAG_SKIP_OPEN_PARENT | OPEN_FLAG_REMOVE_IMAGE);
```

Comment:
And here it move the image to trash, i.e. no actual removal. So I think the flag is not needed here too.